### PR TITLE
os/board/rtl8721csm: support WPA3 128bytes passphrase

### PR DESCRIPTION
**Pull request by gexuyan**

### CODE REVIEW

File: os/board/rtl8721csm/src/component/common/drivers/wlan/realtek/include/wifi_constants.h [hunk 1]
```diff
@@ -53,7 +53,9 @@ extern "C" {
 #define WPA3_SECURITY		0x00800000
 #define WPS_ENABLED     0x10000000
 
-#define RTW_MAX_PSK_LEN		(64)
+#define RTW_WPA2_MAX_PSK_LEN		(64)
```
Comment (jeongchanKim) [inline os/board/rtl8721csm/src/component/common/drivers/wlan/realtek/include/wifi_constants.h:56]: Is this define related to `IW_WPA2_PASSPHRASE_MAX_SIZE`?
If then, how about defining the value with `IW_WPA2_PASSPHRASE_MAX_SIZE` instead of `64`?

**Review by pillip8282 — COMMENTED** [review]
How did you verify passpharase which has more than 64 characters?
TizenRT only supports maximum 64 characters. Could you tell  me how did you verify it?

**Comment by gexuyan** [comment]
> How did you verify passpharase which has more than 64 characters?
> TizenRT only supports maximum 64 characters. Could you tell me how did you verify it?

After changing TASH_LINEBUFLEN from 128 to 256, it should be able to verify with wm_test commands. I have verified with Synology MR2200ac router and it works ok.

File: os/board/rtl8721csm/src/component/common/drivers/wlan/realtek/include/wifi_constants.h (hunk 1 above)
Comment (gexuyan) [reply os/board/rtl8721csm/src/component/common/drivers/wlan/realtek/include/wifi_constants.h:56]: Dear @jeongchanKim ,
This is different, RTW_WPA2_MAX_PSK_LEN is defined for upper layer.

File: apps/shell/tash_internal.h [hunk 2]
```diff
@@ -1,93 +1,93 @@
...
-#endif
-#ifndef CONFIG_DISABLE_SIGNALS
-extern int tash_sleep(int argc, char **args);
-#ifdef CONFIG_TASH_USLEEP
-extern int tash_usleep(int argc, char **args);
-#endif
-#endif
-
-#endif							/* __APPS_SHELL_TASH_INTERNAL_H */
+/****************************************************************************
```
Comment (sunghan-chang) [inline apps/shell/tash_internal.h:1]: You should check something like line end. The format is changed. Please check and revert this.

File: apps/shell/tash_internal.h [hunk 3]
```diff
@@ -1,93 +1,93 @@
...
+
+#ifndef __APPS_SHELL_TASH_INTERNAL_H
+#define __APPS_SHELL_TASH_INTERNAL_H
+#include <tinyara/config.h>
+#include <stdlib.h>
+#include <debug.h>
+
+#define tash_alloc(a)         malloc(a)
+#define tash_free(a)          free(a)
+#define TASH_LINEBUFLEN       (256)
```
Comment (sunghan-chang) [inline apps/shell/tash_internal.h:29]: Could you let me know when this is necessary? This causes more memory consumption.

Comment (gexuyan) [reply apps/shell/tash_internal.h:29]: Dear @sunghan-chang ,
This modification is needed to key in 128bytes length password with wm_test tash command.

File: apps/shell/tash_internal.h (hunk 2 above)
Comment (gexuyan) [reply apps/shell/tash_internal.h:1]: Sure no problem~

File: apps/shell/tash_internal.h (hunk 3 above)
Comment (sunghan-chang) [reply apps/shell/tash_internal.h]: Ok. Even this is necessary to support 128bytes password, this change should be splitted. Please make a separate commit for this. I don't mean new PR, just separte commit.

**Comment by sunghan-chang** [comment]
1. There are conflicts.
2. You should use `git rebase` not `git pull`. Please remove `gexuyan
Merge branch 'SDAIOT_168_WPA3_128bytes_passphrase' of https://github.…` kinds of commits.

**Comment by gexuyan** [comment]
> 1. There are conflicts.
> 2. You should use `git rebase` not `git pull`. Please remove `gexuyan Merge branch 'SDAIOT_168_WPA3_128bytes_passphrase' of https://github.…` kinds of commits.

It seems these two Merge branch 'SDAIOT_168_WPA3_128bytes_passphrase' of https://github.… cannot be removed and it includes the tash_internal.h modifications. I will raise a new PR for this issue. Thank You.
